pkg/client/typed/discovery/discovery_client.go

```diff
@@ -67,6 +67,8 @@ type ServerResourcesInterface interface {
 	ServerResourcesForGroupVersion(groupVersion string) (*unversioned.APIResourceList, error)
 	// ServerResources returns the supported resources for all groups and versions.
 	ServerResources() (map[string]*unversioned.APIResourceList, error)
+	// ServerNamespacedResources returns the supported namespaced resources for all groups and versions.
+	ServerNamespacedResources() ([]unversioned.GroupVersionResource, error)
```

Comment:
Not sure I'd expect a separate method. I'd expect the existing method to return all the groupresources, with versions in server-indicated order (preferred version first)